apis/v1/httproute_types.go

```diff
@@ -119,6 +119,7 @@ type HTTPRouteSpec struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	// +kubebuilder:default={{matches: {{path: {type: "PathPrefix", value: "/"}}}}}
+	// +kubebuilder:validation:XValidation:message="While 16 rules and 64 matches are allowed, the total matches must be less than 128",rule="(self.size() > 0 ? self[0].matches.size() : 0) + (self.size() > 1 ? self[1].matches.size() : 0) + (self.size() > 2 ? self[2].matches.size() : 0) + (self.size() > 3 ? self[3].matches.size() : 0) + (self.size() > 4 ? self[4].matches.size() : 0) + (self.size() > 5 ? self[5].matches.size() : 0) + (self.size() > 6 ? self[6].matches.size() : 0) + (self.size() > 7 ? self[7].matches.size() : 0) + (self.size() > 8 ? self[8].matches.size() : 0) + (self.size() > 9 ? self[9].matches.size() : 0) + (self.size() > 10 ? self[10].matches.size() : 0) + (self.size() > 11 ? self[11].matches.size() : 0) + (self.size() > 12 ? self[12].matches.size() : 0) + (self.size() > 13 ? self[13].matches.size() : 0) + (self.size() > 14 ? self[14].matches.size() : 0) + (self.size() > 15 ? self[15].matches.size() : 0) <= 128"
```

Comment:
This is definitely gross, but the least bad we can offer until CEL cost estimation allows us to use `map` here (Kubernetes 1.30+). Here's the [CEL playground link](https://playcel.undistro.io/?content=H4sIAAAAAAAAA43TzQqCQBDA8VcZ9qS0ieNH6hJ17hnUg%2BhGgVqgXYrevV1qNugDhrnN8ltY%2FuxNtLoXSnjepPt9MB2v2vNhAyFswW7KsA6GZm4PeqJDBaEPi2r8EEgCuSIiEXFFTCLmioREwhUpiZQrViRWXJGRyLgiJ5FzRUGiYBd00ZFf%2FZ2d3R1deGSXR5ce2e3RxUd2fXT58Xd%2FH9bmAbmQomvmZjeeL7P5PlaoalzCSygoUUY1fK0kPgfln7HGXD6cOm3utZ%2Fz%2FgC1SMJupAMAAA%3D%3D) for future readers.